homeassistant/components/iotty/__init__.py

```diff
@@ -0,0 +1,57 @@
+"""The iotty integration."""
+
+from __future__ import annotations
+
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.config_entry_oauth2_flow import (
+    OAuth2Session,
+    async_get_config_entry_implementation,
+)
+from homeassistant.helpers.device_registry import DeviceEntry
+
+from . import coordinator
+from .const import DOMAIN, KNOWN_DEVICES
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up iotty from a config entry."""
+    _LOGGER.debug("async_setup_entry entry_id=%s", entry.entry_id)
+
+    implementation = await async_get_config_entry_implementation(hass, entry)
+    session = OAuth2Session(hass, entry, implementation)
+
+    data_update_coordinator = coordinator.IottyDataUpdateCoordinator(
+        hass, entry, session
+    )
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = data_update_coordinator
+    hass.data[DOMAIN].setdefault(KNOWN_DEVICES, set())
```

Comment:
Please change to use this pattern instead of storing untyped data in `hass.data` https://developers.home-assistant.io/blog/2024/04/30/store-runtime-data-inside-config-entry

Reply:
Hi, I've followed the instructions, I hope that satisfies your request